### PR TITLE
multi-agent reward handlers 

### PR DESCRIPTION
**Pull request by pzhokhov**
a bunch of the java reward handlers compute the reward via subscribing to particular events via eventbus. When such events are generated server-side, reward for multi-agent environments can be misattributed (i.e. all the reward is accumulated by the client in the same java process as server, instead of going to the correct agent)

This PR:
- adds function to RewardBase that lets subclasses to check if the player that generated the reward is the same that instantiated the Reward class.  The logic is as follows - the Reward methods that are annottated with `SubscribeEvent` should either a) forward the reward to client via explicit message OR b) pass the check. 
- Fixes RewardForPossesingItem to send messages on obtaining and loosing the item 
- Deprecates RewardForCollectingItem

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -284,8 +284,8 @@ tests/logs/
 jenkins_home/
 
 #eclipse
-Malmo/Minecraft/*.launch
+minerl/Malmo/Minecraft/*.launch
```
Comment (brandonhoughton) [inline .gitignore:287]: Nice catch =)

File: minerl/Malmo/Minecraft/src/main/java/com/microsoft/Malmo/MissionHandlers/RewardForPossessingItemImplementation.java [hunk 2]
```diff
@@ -267,5 +315,6 @@ public void getReward(MissionInit missionInit, MultidimensionalReward reward) {
     public void cleanup() {
         super.cleanup();
         MinecraftForge.EVENT_BUS.unregister(this);
+        MalmoMod.MalmoMessageHandler.deregisterForMessage(this, MalmoMessageType.SERVER_COLLECTITEM);
```
Comment (brandonhoughton) [inline minerl/Malmo/Minecraft/src/main/java/com/microsoft/Malmo/MissionHandlers/RewardForPossessingItemImplementation.java:318]: Should you deregister SERVER_DISCARDITEM here as well?

**Review by brandonhoughton — CHANGES_REQUESTED** [review]

File: minerl/Malmo/Minecraft/src/main/java/com/microsoft/Malmo/MissionHandlers/RewardForPossessingItemImplementation.java (hunk 2 above)
Comment (pzhokhov) [reply minerl/Malmo/Minecraft/src/main/java/com/microsoft/Malmo/MissionHandlers/RewardForPossessingItemImplementation.java:318]: good point, thanks!

Comment (pzhokhov) [reply minerl/Malmo/Minecraft/src/main/java/com/microsoft/Malmo/MissionHandlers/RewardForPossessingItemImplementation.java:318]: done!

**Review by brandonhoughton — APPROVED** [review]

**Pull request merged**